app/flatpak-quiet-transaction.c

```diff
@@ -91,25 +91,55 @@ new_operation (FlatpakTransaction          *transaction,
     }
 }
 
-static char *
-op_type_to_string (FlatpakTransactionOperationType operation_type)
+static void
+print_op_error_msg (FlatpakTransactionOperationType  operation_type,
+                    FlatpakRef                      *rref,
+                    const char                      *msg,
+                    gboolean                         non_fatal)
 {
+  /* Here we go to great lengths not to split the sentences. See
+   * https://wiki.gnome.org/TranslationProject/DevGuidelines/Never%20split%20sentences
+   */
   switch (operation_type)
     {
     case FLATPAK_TRANSACTION_OPERATION_INSTALL:
-      return _("install");
+      if (non_fatal)
+        g_printerr (_("Warning: Failed to install %s: %s\n"),
+                    flatpak_ref_get_name (rref), msg);
+      else
+        g_printerr (_("Error: Failed to install %s: %s\n"),
+                    flatpak_ref_get_name (rref), msg);
```

Comment:
If `Warning:` and `Error:` are considered prefixes that don't really require additional context, it could be possible to avoid all that repetition and just start the print with a `%s`, similarly to how it was done previously.

Something like:
```c
if (non_fatal)
  prefix = _("Warning");
else
  prefix = _("Error");

[...]

        g_printerr (_("%s Failed to install %s: %s\n"),
                    prefix, flatpak_ref_get_name (rref), msg);
```

Reply:
I brought this point up in the Flatpak matrix channel, and some folks were skeptical whether it is really true that the prefix couldn't affect the translation of the rest of the message in every language (like possibly Japanese) so I suppose it's best to leave the sentence whole.

Reply:
I see, I only speak Italian and English, so I was under the wrong impression that they were probably not affecting the rest of the message.
That's a bit unfortunate but without a better alternative that's good enough.

Reply:
I realized I left "Warning: "/"Error: " as a separate prefix in operation_error() in flatpak-cli-transaction.c so I added another commit for that.

I also noticed that message_handler() adds a "Warning: " prefix to strings but that is its own set of issues (e.g. the g_message() calls in flatpak-transaction.c already have a "Warning: " prefix). See also https://github.com/flatpak/flatpak/pull/4360#issuecomment-941721692